### PR TITLE
Update native publishing AOT documentation

### PR DESCRIPTION
**Pull request by jkotas**
- Move prerequisites into separate section to make them easier to reference from other docs and stand out more
- Add link to NativeAOT samples
- Mention that the samples include Dockerfiles

### CODE REVIEW

**Comment by jkotas** [comment]
cc @dotnet/ilc-contrib

**Review by MichalStrehovsky — APPROVED** [review]
Thank you!

File: docs/core/deploying/native-aot.md [hunk 1]
```diff
@@ -18,41 +18,41 @@ There are some limitations in the .NET native AOT deployment model, with the mai
...
 
     ```sh
     sudo apk add clang gcc lld musl-dev build-base zlib-dev
     ```
 
-03. Publish the app for a specific runtime identifier using `dotnet publish -r <RID>`.
+## Publish Native AOT - CLI
 
-    The following example publishes the app for Windows as a native AOT application on a machine with the required pre-requisites installed.
+01. Add `<PublishAot>true</PublishAot>` to your project file.
```
Comment (gewarren) [inline docs/core/deploying/native-aot.md:43]: Should we add this property to the MSBuild reference at https://docs.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props#compilation-related-properties?

Comment (jkotas) [reply docs/core/deploying/native-aot.md:43]: I do not think it makes sense to document all properties one can possible use in .csproj in one giant file.

https://docs.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props#compilation-related-properties is missing number of properties. For example, it is missing `PublishReadyToRun` and number of other properties documented in https://docs.microsoft.com/en-us/dotnet/core/deploying/ready-to-run; or it is missing `PublishTrimmed` and number of other properties documented in https://docs.microsoft.com/en-us/dotnet/core/deploying/trimming/trimming-options .

**Pull request merged**